pkg/kubectl/service_basic.go

```diff
@@ -209,18 +210,20 @@ func (s ServiceCommonGeneratorV1) StructuredGenerate() (runtime.Object, error) {
 		return nil, err
 	}
 	ports := []api.ServicePort{}
-	for _, tcpString := range s.TCP {
-		port, targetPort, err := parsePorts(tcpString)
+	for _, portsString := range s.Ports {
+		port, targetPort, err := parsePorts(portsString)
 		if err != nil {
 			return nil, err
 		}
 
-		portName := strings.Replace(tcpString, ":", "-", -1)
+		separatorPos := strings.Index(portsString, ":")
+		protocol := strings.ToUpper(portsString[0:separatorPos])
+		portName := strings.Replace(portsString[separatorPos+1:], ":", "-", -1)
```

Comment:
since we support specifying protocol, it'd be good to put protocol into the port name.